lib/recursive-methods.rb

```diff
@@ -1,49 +1,198 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
+# https://github.com/stupendousC/recursion-writing
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+  if (n == 1) || (n == 0)
+    return 1
+  elsif n > 1
+    return n * factorial(n-1)
+  else
+    raise ArgumentError, "argument must be >= 0"
+  end
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def reverse(s)
```

Comment:
👍 
This works, but because you create a new array with each recursive call this is O(n<sup>2</sup>) for both time/space complexity.